lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1411,7 +1411,7 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
         // Handle `copy_addr`.
         else if (auto *cai = dyn_cast<CopyAddrInst>(&inst)) {
           if (isUseful(cai->getDest(), i))
-            setUsefulIfDifferentiable(cai->getSrc(), i);
+            propagateUsefulThroughBuffer(cai->getSrc(), i);
```

Comment:
Sorry for the random change - I found this while investigating https://bugs.swift.org/browse/TF-246.
I think propagation here is correct though.